plugins/modules/foreman_job_template.py

```diff
@@ -249,95 +249,88 @@
-      name: "*"
-      state: absent
+    module: foreman_job_template
+    username: "admin"
+    password: "admin"
+    server_url: "https://foreman.example.com"
+    name: "*"
+    state: absent
 
 - name: "Assign all Job Templates to the same organization(s)"
   local_action:
-      module: foreman_job_template
-      username: "admin"
-      password: "admin"
-      server_url: "https://foreman.example.com"
-      name: "*"
-      state: present
-      organizations:
-      - DALEK INC
-      - sky.net
-      - Doc Brown's garage
+    module: foreman_job_template
+    username: "admin"
+    password: "admin"
+    server_url: "https://foreman.example.com"
+    name: "*"
+    state: present
+    organizations:
+    - DALEK INC
+    - sky.net
+    - Doc Brown's garage
 
 '''
 
 RETURN = ''' # '''
 
-try:
-    from nailgun.entities import (
-        Organization,
-        Location,
-    )
-
-    from ansible.module_utils.ansible_nailgun_cement import (
-        JobTemplate,
-        TemplateInput,
-        find_entities,
-        find_entities_by_name,
-        find_template_input,
-        ForemanEntityAnsibleModule,
-        naildown_entity,
-        naildown_entity_state,
-        sanitize_entity_dict,
-    )
+import os
+from ansible.module_utils.foreman_helper import (
+    ForemanEntityAnsibleModule,
+    parse_template,
+    parse_template_from_file,
+)
 
-    import os
-    from ansible.module_utils.foreman_helper import (
-        parse_template,
-        parse_template_from_file,
-    )
-except ImportError:
-    pass
-
-
-# This is the only true source for names (and conversions thereof)
-name_map = {
-    'audit_comment': 'audit_comment',
-    'description_format': 'description_format',
-    'effective_user': 'effective_user',
-    'job_category': 'job_category',
-    'locations': 'location',
-    'locked': 'locked',
-    'name': 'name',
-    'organizations': 'organization',
-    'provider_type': 'provider_type',
-    'snippet': 'snippet',
-    'template': 'template',
-}
 
 template_defaults = {
     'provider_type': 'SSH',
+    'job_category': 'unknown',
+}
+
+
+template_input_entity_spec = {
+    'name': dict(required=True),
+    'description': dict(),
+    'required': dict(type='bool'),
+    'advanced': dict(type='bool'),
+    'input_type': dict(required=True, choices=[
+        'user',
+        'fact',
+        'variable',
+        'puppet_parameter',
+    ]),
+    'fact_name': dict(),
+    'variable_name': dict(),
+    'variable_name': dict(),
+    'puppet_class_name': dict(),
+    'puppet_parameter_name': dict(),
+    'options': dict(type='list'),
+    'value_type': dict(choices=[
+        'plain',
+        'search',
+        'date',
+    ]),
+    'resource_type': dict(),
 }
 
 
 def main():
     module = ForemanEntityAnsibleModule(
-        argument_spec=dict(
-            # Entity parameter
+        entity_spec=dict(
             audit_comment=dict(),
             description_format=dict(),
```

Comment:
Can we remove this effective user line?

Reply:
sure. A quick look at the api shows, that its implementation is at least not that easy.